test/src/test/java/org/apache/accumulo/test/functional/TableChangeStateIT.java

```diff
@@ -0,0 +1,549 @@
+        assert (elt.getKey().getRow().toString().equals(expected));
+        count++;
+      }
+
+      log.trace("Scan time for {} rows {} ms", NUM_ROWS, TimeUnit.MILLISECONDS.convert((System.nanoTime() - startTimestamp), TimeUnit.NANOSECONDS));
+
+      scanner.close();
+
+      if (count != NUM_ROWS) {
+        throw new IllegalStateException(String.format("Number of rows %1$d does not match expected %2$d", count, NUM_ROWS));
+      }
+    } catch (AccumuloException | AccumuloSecurityException | TableNotFoundException | TableExistsException ex) {
+      throw new IllegalStateException("Create data failed with exception", ex);
+    }
+  }
+
+  /**
+   * Provides timing information and online command status results for online command executed asynchronously. The
+   */
+  private static class OnlineStatus {
+
+    private long started = 0L;
+    private long completed = 0L;
+
+    // set to true - even before running so that we can block on isRunning before thread may have started.
+    private final AtomicBoolean isRunning = new AtomicBoolean(Boolean.TRUE);
+    private Boolean hasError = Boolean.FALSE;
+    private Exception exception = null;
+
+    /**
+     * Returns true until operation has completed.
+     *
+     * @return false when operation has been set complete.
+     */
+    Boolean isRunning() {
+      return isRunning.get();
+    }
+
+    /**
+     * start timing operation.
+     */
+    void setRunning() {
+      isRunning.set(Boolean.TRUE);
+      started = System.nanoTime();
+    }
+
+    /**
+     * stop timing and set completion flag.
+     */
+    void setComplete() {
+      completed = System.nanoTime();
+      isRunning.set(Boolean.FALSE);
+    }
+
+    /**
+     * Returns true if setError was called to complete operation.
+     *
+     * @return true if an error was set, false otherwise.
+     */
+    boolean hasError() {
+      return hasError;
+    }
+
+    /**
+     * Returns exception if set by {@code setError}.
+     *
+     * @return the exception set with {@code setError} or null if not set.
+     */
+    public Exception getException() {
+      return exception;
+    }
+
+    /**
+     * Marks the operation as complete, stops timing the operation, with error status and exception that caused the failure.
+     *
+     * @param ex
+     *          the exception that caused failure.
+     */
+    public void setError(Exception ex) {
+      hasError = Boolean.TRUE;
+      exception = ex;
+      setComplete();
+    }
+
+    /**
+     * @return running time in nanoseconds.
+     */
+    long runningTime() {
+      return completed - started;
+    }
+  }
+
+  /**
+   * Run online operation in a separate thread. The operation status can be tracked with {@link OnlineStatus} returned by {@link OnLineThread#getStatus}. The
+   * operation timing is started when run is called. If an exception occurs, it is available in the status.
+   */
+  private class OnLineThread implements Runnable {
```

Comment:
Using FutureTask may simplify this class... or it may not.  I am not sure, its just something I thought of while looking at this code.